content/docs/vibe-coding/al-naming-conventions.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Naming Conventions Rules"
+description: >
+  Comprehensive naming conventions for AL files, objects, variables, and functions
+globs: ["*.al"]
+alwaysApply: true
+---
+
+# Naming Conventions Rules
+
+Consistent naming conventions improve code readability, maintainability, and help AI assistants understand code structure and intent.
+
+## Rule 1: Object Naming Conventions
+
+### Intent
+Use consistent naming patterns for all AL objects to improve discoverability and maintain professional standards. Use PascalCase for object names (tables, pages, reports, codeunits) and meaningful, descriptive names that clearly indicate the object's purpose. Object names must not exceed 30 characters total, with a maximum of 26 characters for the name itself to reserve space for prefixes/affixes (3 characters + 1 space).
+
+### Examples
+
+```al
+// Good examples (within 26 character limit)
+table 50100 "Customer Ledger Entry"     // 20 chars
+page 50101 "Sales Invoice"              // 13 chars  
+codeunit 50102 "Sales Invoice Posting"  // 21 chars
+report 50103 "Customer Statement"       // 18 chars
+```
+
+```al
+// Bad examples (avoid abbreviations, unclear names, or length violations)
+table 50100 "CustLE"                           // Too abbreviated
+page 50101 "SalesInv"                          // Too abbreviated
+table 50104 "Very Long Customer Ledger Entry" // 32 chars - exceeds limit
+codeunit 50102 "SIPoster"                      // Unclear abbreviation
+```
+
+## Rule 2: File Naming Conventions
+
+### Intent
+Establish consistent file naming patterns that clearly identify object types and facilitate organized development. Use pattern `<ObjectName>.<ObjectType>.al` and maintain consistency across all file names. Ensure file names are descriptive and match the AL object name within the files.
+
+### Examples
+
+```al
+// Good examples
+NoSeries.Page.al
+NoSeries.Table.al
+NoSeriesErrorsImpl.Codeunit.al
+NoSeriesSetup.Codeunit.al
+CustomerCard.Page.al
+SalesHeader.Table.al
+PostSalesInvoice.Codeunit.al
+ItemLedgerEntry.Report.al
+InventorySetup.PageExt.al
+SalesHeader.TableExt.al
+
+// For implementations and interfaces
+INoSeries.Interface.al
+NoSeriesImpl.Codeunit.al
+
+// For test files
+NoSeriesTests.Codeunit.al
+SalesPostingTests.Codeunit.al
+```
+
+## Rule 3: Variable and Function Naming
+
+### Intent
+Use consistent naming conventions for variables and functions to improve code readability. Use camelCase for variable and function names, descriptive names that clearly indicate purpose, and avoid abbreviations unless they are well-known business terms. Use consistent parameter naming in procedures.
```

Comment:
Is camelCase a new convention? The current convention is much more the PascalCase. It is also in contradiction with the case of the event subscribers (Rule 4).